helix-core/src/test/java/org/apache/helix/integration/task/TestNoDoubleAssign.java

```diff
@@ -158,9 +158,11 @@ private void breakConnection() {
     _executorServiceConnection = Executors.newScheduledThreadPool(THREAD_COUNT);
     _executorServiceConnection.scheduleAtFixedRate(() -> {
       // Randomly pick a Participant and cause a transient connection issue
-      int participantIndex = RANDOM.nextInt(_numNodes);
-      _participants[participantIndex].disconnect();
-      startParticipant(participantIndex);
+      synchronized (this){
```

Comment:
In this case, shall we lock all the start/stop callers?
I think a better way is to lock inside the start and stop methods. Then add a reset method in addition which requests the same lock.